apps/web/src/components/widget/AvatarContainer.tsx

```diff
@@ -0,0 +1,194 @@
+    type: SystemAvatarIconEnum.INFO,
+    bgColor: '#0000FF',
+  },
+  {
+    icon: <UpCircleFilled />,
+    type: SystemAvatarIconEnum.UP,
+    bgColor: colors.B70,
+  },
+  {
+    icon: <QuestionCircleFilled />,
+    type: SystemAvatarIconEnum.QUESTION,
+    bgColor: colors.B70,
+  },
+  {
+    icon: <CheckCircleFilled />,
+    type: SystemAvatarIconEnum.SUCCESS,
+    bgColor: colors.success,
+  },
+  {
+    icon: <CloseCircleFilled />,
+    type: SystemAvatarIconEnum.ERROR,
+    bgColor: colors.error,
+  },
+];
+
+const AvatarContainer = ({ value, onChange }: { onChange: (data: any) => void; value: IAvatarDetails }) => {
+  const [opened, setOpened] = useState(false);
+  const [tooltipOpened, setTooltipOpened] = useState(() => {
+    return value.type === AvatarTypeEnum.NONE;
+  });
+
+  const { classes, theme } = useStyles();
+  const dark = theme.colorScheme === 'dark';
+
+  function handleAvatarPopover() {
+    setOpened((prev) => !prev);
+    if (tooltipOpened) {
+      setTooltipOpened(false);
+    }
+  }
+
+  return (
+    <>
+      <Popover
+        target={
+          <Tooltip label={<TooltipLabel />} position="left" opened={tooltipOpened}>
+            <AvatarWrapper onClick={handleAvatarPopover} dark={dark}>
+              <RenderAvatar avatarDetails={value} />
+            </AvatarWrapper>
+          </Tooltip>
+        }
+        opened={opened}
+        position="bottom"
+        placement="start"
+        withArrow
+        classNames={{
+          inner: classes.inner,
+          target: classes.target,
+          arrow: classes.arrow,
+          body: classes.body,
+          popover: classes.popover,
+        }}
+      >
+        <Stack>
+          <Group align="center">
+            <Box>
+              <Avatar />
+            </Box>
+            <Box>
+              <Text>User avatar</Text>
+            </Box>
+            <Box
+              sx={{
+                marginLeft: 'auto',
+              }}
+            >
+              <Switch
+                checked={value.type === AvatarTypeEnum.USER}
+                onChange={(event) =>
+                  onChange({
+                    type: event.currentTarget.checked ? AvatarTypeEnum.USER : AvatarTypeEnum.NONE,
+                    data: null,
+                  })
+                }
+              />
+            </Box>
+          </Group>
+          <Divider label={<Text color={colors.B60}>Platform/System Avatars</Text>} labelPosition="left" />
+
+          <Group align="center">
+            <Box>
+              <Camera />
+            </Box>
+            <Box>
+              <Text>Enter Avatar's URL</Text>
+            </Box>
+          </Group>
+          <Input
+            placeholder="Enter Avatar's URL"
+            onChange={(event) =>
+              onChange({
+                data: event.target.value,
+                type: event.target.value ? AvatarTypeEnum.SYSTEM_CUSTOM : AvatarTypeEnum.NONE,
+              })
+            }
```

Comment:
I'm thinking if we should do `onBlur` instead the `onChange` here because rn it triggers the request every time I type something and because of that it blinks.

https://user-images.githubusercontent.com/2607232/200319806-7cc0b28f-5d5a-4d05-bcb7-702172dffabf.mov